### ISSUE TITLE
Don't create .openjfx cache folder in user directory?

### ISSUE THREAD

**Issue by CharlesVilliers** [JabRef/jabref#5817]
Not sure if bug, question or feature request: With the most recent master of JabRef 5 I am slightly irritated not only by the appearing command window (this issue depends on externals), but also by the folder .openjfx that is created in my user directory. This has not been the case with previous versions. I only have very few folders in this directory - could the cache not be stored somewhere else or at least hidden?
Thanks for your work!

**Comment by tobiasdiez** [comment]
As the name suggests, this folder is created by OpenFX. 
The thread https://stackoverflow.com/questions/53382810/configure-openjfx-11-to-extract-its-dlls-into-a-different-user-specified-directo suggests that adding `-Djavafx.cachedir=C:\\Users\\xxx\\Temp\\cache` to the last line (before `-m`) in `JabRef\runtime\bin\JabRef.bat` let you choose a different folder.

I agree that this is folder can be irrating. I would suggest you open a bug report at https://bugreport.java.com/bugreport/. Closing this issue as we can sadly do nothing.

**Comment by arnaumas** [comment]
Is there a way to add this setting on macOS? The equivalent of Jabref.bat according to the [docs](https://docs.jabref.org/advanced/commandline) is an executable which cannot be edited.

Also, JabRef is creating .pdf.cache in my home directory, which I would also prefer moving to a dedicated cache directory.

**Comment by Siedlerchr** [comment]
Maybe it can be added to the cfg

**Comment by technobirb** [comment]
Could the paths be changed on Linux to use XDG directories as well?
Currently they are placed in home even for [flatpak](https://discourse.flathub.org/t/use-home-folder-rather-than-var-for-config-files/7926/8): https://github.com/flathub/org.jabref.jabref/issues/37

**Comment by Siedlerchr** [comment]
@technobirb  you can modify the [JavaOptions] section in the .cfg file. You can also place an additional config file in your system https://bugs.openjdk.org/browse/JDK-8287060 

@arnaumas  This works also for mac

(make sure to copy JabRef's one with all the options

The other idea is to probably switch the variable for the user directroy:
```

[JavaOptions]
java-options=-Djavafx.cachedir=/Path/To/Your/direcotry
java-options=-Duser.home=/path/to/my/new/home/dir
```

**Comment by koppor** [comment]
@Siedlerchr Shouldnt this become a FAQ?

**Comment by technobirb** [comment]
Tbh there are [arguments](https://github.com/NationalSecurityAgency/ghidra/issues/908) to be had on using [XDG directories](https://specifications.freedesktop.org/basedir-spec/latest/) on Linux (and [APPDATA on Windows](https://learn.microsoft.com/en-us/answers/questions/584279/whats-the-idea-behind-the-name-of-the-appdata-fold)), regardless whether it's flatpak or not, though it's more important for flatpak. Littering $HOME isn't great. There is a [discussion](https://bugzilla.redhat.com/show_bug.cgi?id=1154277#c9) in Fedora to change this, but I don't like [their chances](https://bugs.openjdk.org/browse/JDK-8290140?focusedId=14510115&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-14510115). Alternatively, apps can use different paths [themselves](https://github.com/JetpackDuba/Gitnuro/issues/112).

**Comment by koppor** [comment]
I think we should generally pass the java options parameter for Linux to a proper XDG directory.

Maybe also on Windows for APPDATA (if it isn't used yet)

**Comment by Siedlerchr** [comment]
I don't think we should mess with the java options. This will create more problems than help anyone as it's just for maybe 0.5% of users relevant who care where there data is located.

**Comment by koppor** [comment]
Seeing the linked https://bugzilla.redhat.com/show_bug.cgi?id=1154277#c9, I think, this could be included in OpenJDK eventually?

Maybe someone has a https://bugzilla.redhat.com account and ask to work on an OpenJDK patch (to fix the linked https://bugs.openjdk.org/browse/JDK-8290140).

**Comment by Siedlerchr** [comment]
Something for your to look into @Maran23

**Comment by Maran23** [comment]
This can be solved by 2 different approaches:
- The binary is built with JMods, then this is not needed
- The binary is built with GraalVM, then this is not needed aswell
